geode-docs/developing/outside_data_sources/implementing_data_loaders.html.md.erb

```diff
@@ -19,70 +19,156 @@ See the License for the specific language governing permissions and
-            <parameter name="URL">
-                <string>jdbc:cloudscape:rmi:MyData</string>
-            </parameter>
-        </cache-loader>
-        ```
-    -   Java:
-
-        ``` pre
-        RegionFactory<String,Object> rf = cache.createRegionFactory(REPLICATE);
-        rf.setCacheLoader(new QuoteLoader());
-        quotes = rf.create("NASDAQ Quotes");
-        ```
-
-**Note:**
-You can also configure Regions using the gfsh command-line interface, however you cannot configure a `cache-loader` using gfsh. See [Region Commands](../../tools_modules/gfsh/quick_ref_commands_by_area.html#topic_EF03119A40EE492984F3B6248596E1DD).
+}
+```
 
+If you need to run `Region` API calls from your implementation,
+spawn separate threads for them.
+Do not make direct calls to `Region` methods from your `load` method,
+as it could cause the cache loader to block,
+hurting the performance of the distributed system.
+
+## Configure and Deploy
+
+Use one of these three ways to configure and deploy the cache loader:
+
+**Option 1:** If configuring a cluster by defining a `cache.xml` file,
+deploy by adding the cache loader to the classpath when starting servers.
+
+Here is an example configuration within the `cache.xml` file
+that specifies the loader without parameters:
+
+``` pre
+<region-attributes>
+    <cache-loader>
+        <class-name>myLoader</class-name>
+    </cache-loader>
+</region-attributes>
+```
+Or, here is an example configuration within the `cache.xml` file
+that specifies the loader with parameters:
+
+``` pre
+<cache-loader>
+    <class-name>com.company.data.DatabaseLoader</class-name>
+    <parameter name="URL">
+        <string>jdbc:cloudscape:rmi:MyData</string>
+    </parameter>
+</cache-loader>
+```
+
+To deploy the JAR file,
+add the cache loader JAR file to the classpath when starting all servers.
+For example:
+
+``` pre
+gfsh>start server --name=s2 --classpath=/var/data/lib/myLoader.jar
+```
+
+**Option 2:** 
+If deploying the JAR file at server startup,
+add the JAR file to the classpath and use gfsh to apply the
+configuration to the region.
+
+To deploy the JAR file,
+add the cache loader JAR file to the classpath when starting all
+servers.
+For example:
+
+``` pre
+gfsh>start server --name=s2 --classpath=/var/data/lib/myLoader.jar
+```
+
+Use gfsh to apply the configuration of the `CacheLoader` implementation
+to the region with `gfsh create region` or `gfsh alter region`.
+Here is an example of region creation:
+
+``` pre
+gfsh>create region --name=r3 --cache-loader=com.example.appname.myCacheLoader
+```
+
+Here is an example of altering a region:
+
+``` pre
+gfsh>alter region --name=r3 --cache-loader=com.example.appname.myCacheLoader
+```
+
+**Option 3:** 
+If deploying the JAR file with the gfsh deploy command
+after servers have been started,
+use gfsh to apply the configuration to the region.
+
+After server creation use gfsh to deploy the JAR file to all the
```

Comment:
because deploy command deploys to all servers we should probably state the best practice here? For PR its okay to have loader on every member that hosts the region vs for replicate only one loader across the members?